book/src/dev/rfcs/xxxx-block-subsidy.md

```diff
@@ -0,0 +1,195 @@
+
+To do the calculations and checks the following constants and functions need to be introduced:
+
+## Constants
+
+- `SLOW_START_INTERVAL`
+- `SLOW_START_SHIFT`
+- `MAX_BLOCK_SUBSIDY`
+- `PRE_BLOSSOM_POW_TARGET_SPACING`
+- `POST_BLOSSOM_POW_TARGET_SPACING`
+- `BLOSSOM_POW_TARGET_SPACING_RATIO`
+- `PRE_BLOSSOM_HALVING_INTERVAL`
+- `POST_BLOSSOM_HALVING_INTERVAL`
+- `FOUNDERS_FRACTION_DIVISOR`
+- `FOUNDER_ADDRESS_CHANGE_INTERVAL`
+- `FOUNDERS_REWARD_ADDRESSES_MAINNET`
+- `FOUNDERS_REWARD_ADDRESSES_TESTNET`
+
+### Funding streams parameter constants
+
+The design suggests to implement the parameters needed for funding streams as:
+
+```
+/// The funding stream receiver categories
+pub enum FundingStreamsReceivers {
+    /// Electric Coin Company
+    ECC,
+    /// ZCash Foundation
+    ZF,
+    /// Major Grants
+    MG,
+}
+
+/// The numerator for each funding stream receiving category 
+/// as described in [protocol specification §7.9.1][7.9.1].
+/// 
+/// [7.9.1]: https://zips.z.cash/protocol/protocol.pdf#zip214fundingstreams
+const FUNDINGSTREAMS_RECEIVERS_NUMERATORS: &[(i32, FundingStreamsReceivers)] = &[
+    (7, FundingStreamsReceivers::ECC),
+    (5, FundingStreamsReceivers::ZF),
+    (8, FundingStreamsReceivers::MG),
+];
+
+/// Denominator as described in [protocol specification §7.9.1][7.9.1].
+/// 
+/// [7.9.1]: https://zips.z.cash/protocol/protocol.pdf#zip214fundingstreams
+pub const FUNDINGSTREAM_DENOMINATOR: i32 = 100;
+
+/// Height intervals for funding streams on each network.
+pub enum FundingStreamsHeightIntervals {
+    /// For the Mainnet
+    Mainnet,
+    /// For the Testnet
+    Testnet,
+}
+
+/// Start and end Heights for funding streams 
+/// as described in [protocol specification §7.9.1][7.9.1].
+/// 
+/// [7.9.1]: https://zips.z.cash/protocol/protocol.pdf#zip214fundingstreams
+const FUNDINGSTREAMS_HEIGHT_INTERVALS: &[(Height, Height, FundingStreamsHeightIntervals)] = &[
+    (Height(1_046_400), Height(2_726_400), FundingStreamsHeightIntervals::Mainnet),
+    (Height(1_028_500), Height(2_796_000), FundingStreamsHeightIntervals::Testnet),
+];
+```
+
+## General subsidy
+
+The block subsidy and miner reward among other utility functions are inside the general subsidy category.
+
+- `block_subsidy(Height, Network) -> Result<Amount<NonNegative>, Error>` - Total block subsidy.
+- `miner_subsidy(Height, Network) -> Result<Amount<NonNegative>, Error>` - Miner portion.
+- `transaction_fees(&Block) -> Result<Amount<NonNegative>, Error>` - Sum of all the transaction fees.
+- `coinbase_sum_outputs(&Transaction) -> Result<Amount<NonNegative>, Error>` - Sum of all output values in the coinbase transaction.
+- `find_output_with_amount(&Transaction, Amount<NonNegative>) -> Vec<transparent::Output>` - Outputs where value equal to Amount.
+
+## Founders reward
+
+Only functions specific to calculation of founders reward.
```

Comment:
Let's use this link in this RFC, as well as links to the relevant ZIPs:
https://zips.z.cash/protocol/canopy.pdf#foundersreward